# Day 01-  Uber Rides Data Exploration
# Dataset: Uber rides NYC April 2014
# Goal: Load, explore, extract datetime features, observe patterns

In [1]:
import pandas as pd

In [5]:
url = 'https://raw.githubusercontent.com/plotly/datasets/master/uber-rides-data1.csv'

df = pd.read_csv(url, nrows=50000)
print(df.shape)

(50000, 3)


In [34]:
df.head(10)

,Date/Time,Lat,Lon,hour,day,month,year
0,2014-04-01 00:11:00,40.7690,-73.9549,0,Tuesday,4,2014
1,2014-04-01 00:17:00,40.7267,-74.0345,0,Tuesday,4,2014
2,2014-04-01 00:21:00,40.7316,-73.9873,0,Tuesday,4,2014
3,2014-04-01 00:28:00,40.7588,-73.9776,0,Tuesday,4,2014
4,2014-04-01 00:33:00,40.7594,-73.9722,0,Tuesday,4,2014
5,2014-04-01 00:33:00,40.7383,-74.0403,0,Tuesday,4,2014
6,2014-04-01 00:39:00,40.7223,-73.9887,0,Tuesday,4,2014
7,2014-04-01 00:45:00,40.7620,-73.9790,0,Tuesday,4,2014
8,2014-04-01 00:55:00,40.7524,-73.9960,0,Tuesday,4,2014
9,2014-04-01 01:01:00,40.7575,-73.9846,1,Tuesday,4,2014


In [28]:
print("shape:", df.shape)
print("\nColumn names:", df.columns.tolist())
print("\nData Types:\n",df.dtypes)
print("\nNull Counts:\n",df.isnull().sum())
print("\nFirst 5 Rows:")
df.head(5)
print("\nDuplicates:", df.duplicated().sum())

shape: (50000, 7)

Column names: ['Date/Time', 'Lat', 'Lon', 'hour', 'day', 'month', 'year']

Data Types:
 Date/Time    datetime64[ns]
Lat                 float64
Lon                 float64
hour                  int32
day                  object
month                 int32
year                  int32
dtype: object

Null Counts:
 Date/Time    0
Lat          0
Lon          0
hour         0
day          0
month        0
year         0
dtype: int64

First 5 Rows:

Duplicates: 1324


- Dataset has 50k rows and 7 columns
- No null values found in any column
- 1324 Duplicates found
- Data is from April 2014

In [26]:
df['Date/Time']=pd.to_datetime(df['Date/Time'])

df['hour'] = df['Date/Time'].dt.hour
df['day'] = df['Date/Time'].dt.day_name()
df['month'] = df['Date/Time'].dt.month
df['year'] = df['Date/Time'].dt.year

print(df.head())
print("\n")
print(df.dtypes)

            Date/Time      Lat      Lon  hour      day  month  year
0 2014-04-01 00:11:00  40.7690 -73.9549     0  Tuesday      4  2014
1 2014-04-01 00:17:00  40.7267 -74.0345     0  Tuesday      4  2014
2 2014-04-01 00:21:00  40.7316 -73.9873     0  Tuesday      4  2014
3 2014-04-01 00:28:00  40.7588 -73.9776     0  Tuesday      4  2014
4 2014-04-01 00:33:00  40.7594 -73.9722     0  Tuesday      4  2014


Date/Time    datetime64[ns]
Lat                 float64
Lon                 float64
hour                  int32
day                  object
month                 int32
year                  int32
dtype: object


,0
Lat,40.744262
Lon,-73.979859


In [29]:
#basic stat on location
print(df[['Lat','Lon']].describe())

                Lat           Lon
count  50000.000000  50000.000000
mean      40.744262    -73.979859
std        0.035486      0.054782
min       40.225900    -74.703900
25%       40.726000    -73.998700
50%       40.747400    -73.984600
75%       40.763000    -73.971000
max       41.373000    -72.923700


In [39]:
mean = df[['Lat', 'Lon']].mean()
std = df[['Lat', 'Lon']].std()

outliers = df[
    ((df[['Lat', 'Lon']] < (mean - 3 * std)) |
     (df[['Lat', 'Lon']] > (mean + 3 * std)))
    .any(axis=1)
]

print(f"Outliers found: {len(outliers)}")

Outliers found: 2373


- Lat values range from 40.225900 to 373000
- Lon values range from -74.703900 to - 72.923700
- 2373 Outliers in location data

In [41]:
# Busiest Hours
print(df.groupby('hour').size().sort_values(ascending=False).head(5))

hour
17    4167
16    3822
18    3776
19    3346
15    3243
dtype: int64


In [43]:
#Days with most ride
print(df.groupby('day').size().sort_values(ascending=False).head(3))

day
Wednesday    11470
Thursday     10992
Tuesday       9609
dtype: int64


- Most rides happen at 17th hour
- Least rides at 2nd hour
- Wednesday has most rides: 11470
- Sunday has least rides: 3066